### PR TITLE
charm: implement postgres relation

### PR DESCRIPTION
**Pull request by rogpeppe**

### CODE REVIEW

**Comment by jujugui** [comment]
Refer to this link for build results (access rights to CI server needed): 
http://ci-gce.jujugui.org:8080//job/candid/45/

File: charm/Makefile [hunk 1]
```diff
@@ -6,8 +6,20 @@ hooks/charmhelpers: charm-helpers.yaml
 
 .PHONY: lint
 lint:
-	flake8 hooks/config-changed hooks/db-relation-changed hooks/install.py hooks/jaascharm.py hooks/nrpe-relation-changed hooks/status.py hooks/stop hooks/update-status hooks/upgrade-charm hooks/website-relation-joined
-
+	@# ls -d hooks/* | grep -v -E '/(install|charmhelpers)$'
```
Comment (mhilton) [inline charm/Makefile]: Is this line doing anything now?

File: charm/metadata.yaml [hunk 2]
```diff
@@ -19,8 +19,10 @@ provides:
   prometheus:
     interface: prometheus
 requires:
-  db:
+  mongodb:
     interface: mongodb
+  postgres:
+    interface: pgsql
   # TODO allow postgres relation too?
```
Comment (mhilton) [inline charm/metadata.yaml]: TODO is TODONE

File: charm/hooks/postgres-relation-changed [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+            'connection-string': connection_str,
+        },
+    }
+    jaascharm.update_config_and_restart(config)
+
+
+if __name__ == '__main__':
+    hookenv.log('db-relation-changed')
+    relation_changed()
+    jaascharm.update_status(failed_status=charm_status)
```
Comment (mhilton) [inline charm/hooks/postgres-relation-changed]: I'd be expecting charm_status to change too.

File: charm/Makefile (hunk 1 above)
Comment (rogpeppe) [reply charm/Makefile]: It's just so the command is easily available to run to keep the lines below in sync.

File: charm/hooks/postgres-relation-changed (hunk 3 above)
Comment (rogpeppe) [reply charm/hooks/postgres-relation-changed]: good point. done.

File: charm/metadata.yaml (hunk 2 above)
Comment (rogpeppe) [reply charm/metadata.yaml]: TODONE done

**Comment by jujugui** [comment]
Refer to this link for build results (access rights to CI server needed): 
http://ci-gce.jujugui.org:8080//job/candid/46/

File: charm/hooks/status.py [hunk 4]
```diff
@@ -6,10 +6,10 @@ def charm_status(msg):
...
         return 'blocked', msg
-    if hookenv.relation_ids('db'):
-        # We've got a mongodb relation, but we're still getting
+    if hookenv.relation_ids('mongodb') or hookenv.relation_ids('postgres'):
+        # We've got a database relation, but we're still getting
         # an error with "storage" in it. That's probably
-        # because the mongodb application at the other
+        # because the application at the other
         # end of the relation hasn't set its attributes yet.
         return 'waiting', 'waiting for db relation'
```
Comment (mhilton) [inline charm/hooks/status.py]: It might be worth checking these individually so that the message matches the ones in the relation-changed hooks?

Comment (rogpeppe) [reply charm/hooks/status.py]: good point. done.

**Comment by jujugui** [comment]
Refer to this link for build results (access rights to CI server needed): 
http://ci-gce.jujugui.org:8080//job/candid/47/

**Comment by rogpeppe** [comment]
:shipit:

**Comment by jujugui** [comment]
Status: merge request accepted. Url: http://ci-gce.theblues.io:8080/job/candid-merge

**Pull request merged**